refactor(language_server): make linter independent of `Backend`

Sysix opened the pull request:
This PR will refactor the language server, so the Linter is independent of the language server.
The changes are needed to support multiple workspaces in the upstream PR. 
There, every linter will get its own root Uri, and will report back to the `Backend`.

The `Backend` collects the different outputs from the workers and return them to the client when needed.

Sysix commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/10497?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#10497** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/10497?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/10497?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#10479** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/10479?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=10497).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

camchenry commented on crates/oxc_language_server/src/worker.rs:
```diff
@@ -0,0 +1,492 @@
+}
+
+#[cfg(test)]
+mod tests {
```
I think that would be ideal, and probably better reflects actual usage (because users are just loading the directory into a workspace and expect everything to work from there)

camchenry reviewed (approved):
I need to get a better handle on the language server protocol, but I think this makes sense

Sysix committed:
refactor(language_server): make linter independent of `Backend` (#10497)

This PR will refactor the language server, so the Linter is independent of the language server.
The changes are needed to support multiple workspaces in the upstream PR.
There, every linter will get its own root Uri, and will report back to the `Backend`.

The `Backend` collects the different outputs from the workers and return them to the client when needed.
